packages/sdk/src/modules/did/index.ts

```diff
@@ -152,24 +162,44 @@ export class Did extends Base {
     return `did:peaq:${address}`;
   }
 
-  private _createVerificationMethod(address: Address) {
+  private _createVerificationMethod(verification: Verification, address: Address) {
     const id = uuidv4();
     const verificationMethod = new peaqDidProto.VerificationMethod();
 
+    // does this id change if it is Ed25519 vs Sr25519?
     verificationMethod.setId(id);
-    verificationMethod.setType(
-      peaqDidProto.VerificationType.ED25519VERIFICATIONKEY2020
-    );
+
+    if (verification.type !== peaqDidProto.VerificationType.ED25519VERIFICATIONKEY2020 &&
+      verification.type !== peaqDidProto.VerificationType.SR25519VERIFICATIONKEY2020) {
+      throw new Error('Invalid type: Type must be either 0: ED25519VERIFICATIONKEY2020 or 1: SR25519VERIFICATIONKEY2020');
+    }
+
+    verificationMethod.setType(verification.type);
     verificationMethod.setController(this._getDidId(address));
     verificationMethod.setPublickeymultibase(`z${address}`);
 
     return { verificationMethod, verificationId: id };
   }
 
-  private _createService(service: DocumentService) {
+  // added logic to add signature
+  private _createSignature(signature: Signature) {
+    if (!Object.values(peaqDidProto.VerificationType).includes(signature.type)) throw new Error('Signature Type is required');
+    if (!signature.issuer) throw new Error('Signature Issuer is required');
+    if (!signature.hash) throw new Error('Signature Hash is required');
+
+    const signatureMethod = new peaqDidProto.Signature();
+
+    signatureMethod.setType(signature.type);
+    signatureMethod.setIssuer(signature.issuer);
+    signatureMethod.setHash(signature.hash);
+
+    return signatureMethod;
+  }
+
+  private _createService(service: Service) {
     if (!service.id) throw new Error('Service ID is required');
     if (!service.type) throw new Error('Service type is required');
-    if (!('serviceEndpoint' in service) && !('data' in service))
+    if (!(service.serviceEndpoint) && !(service.data))
```

Comment:
If there is no serviceEndpoint or data field set in the service customDocumentFields, this will throw an error